File: src/dotnet/Rider.Plugins.MonoGame/Effect/Psi/EffectLanguage.cs
using JetBrains.Annotations;
using JetBrains.ReSharper.Psi;

namespace Rider.Plugins.MonoGame.Effect.Psi;

[LanguageDefinition(Name)]
public class EffectLanguage : KnownLanguage
{
    public new const string Name = "MGFX";

    [CanBeNull, UsedImplicitly]
    public static EffectLanguage Instance { get; private set; }

    public EffectLanguage()
        : base(Name, "MGFX")
    {
    }
}

File: src/rider/main/kotlin/me/seclerp/rider/plugins/monogame/effect/EffectLanguage.kt
package me.seclerp.rider.plugins.monogame.effect

import com.jetbrains.rider.ideaInterop.fileTypes.RiderLanguageBase

object EffectLanguage : RiderLanguageBase("MGFX", "MGFX") {
    override fun isCaseSensitive(): Boolean = false
}

